pkg/asset/manifests/openshift.go

```diff
@@ -280,6 +280,37 @@ func (o *Openshift) Generate(ctx context.Context, dependencies asset.Parents) er
 		assetData["99_baremetal-provisioning-config.yaml"] = applyTemplateData(baremetalConfig.Files()[0].Data, bmTemplateData)
 	}
 
+	if platform == azuretypes.Name && installConfig.Config.CredentialsMode != types.ManualCredentialsMode {
+		// config is used to created compatible secret to trigger azure cloud
+		// controller config merge behaviour
+		// https://github.com/openshift/origin/blob/90c050f5afb4c52ace82b15e126efe98fa798d88/vendor/k8s.io/legacy-cloud-providers/azure/azure_config.go#L83
+		session, err := installConfig.Azure.Session()
+		if err != nil {
+			return err
+		}
+		config := struct {
+			AADClientID     string `json:"aadClientId" yaml:"aadClientId"`
```

Comment:
I don't understand how these get populated? But also I'm not too familiar with the code :)

Reply:
Marking this as resolved--I completely changed the approach (as this one did not work!)